functional/FunctionalTest.java

```diff
@@ -3633,66 +3621,49 @@ public static void getDefaultRetention_test() throws Exception {
+                GetDefaultRetentionArgs.builder().bucket(bucketName).build());
 
-        if ((!(config.duration().unit() == expectedConfig.duration().unit()
-                && config.duration().duration() == expectedConfig.duration().duration()))
-            || (config.mode() != expectedConfig.mode())) {
+        if (config.mode() != expectedConfig.mode()) {
           throw new Exception(
-              "[FAILED] Expected: expected duration : "
-                  + expectedConfig.duration()
-                  + ", got: "
-                  + config.duration()
-                  + " expected mode :"
-                  + expectedConfig.mode()
-                  + ", got: "
-                  + config.mode());
+              "[FAILED] mode: expected: " + expectedConfig.mode() + ", got: " + config.mode());
+        }
+
+        if (config.duration().unit() != expectedConfig.duration().unit()
+            || config.duration().duration() != expectedConfig.duration().duration()) {
+          throw new Exception(
+              "[FAILED] duration: " + expectedConfig.duration() + ", got: " + config.duration());
         }
 
         expectedConfig =
             new ObjectLockConfiguration(RetentionMode.GOVERNANCE, new RetentionDurationYears(1));
-        client.setDefaultRetention(bucketName, expectedConfig);
-        config = client.getDefaultRetention(bucketName);
+        client.setDefaultRetention(
+            SetDefaultRetentionArgs.builder().bucket(bucketName).config(expectedConfig).build());
+        config =
+            client.getDefaultRetention(
+                GetDefaultRetentionArgs.builder().bucket(bucketName).build());
 
-        if ((!(config.duration().unit() == expectedConfig.duration().unit()
-                && config.duration().duration() == expectedConfig.duration().duration()))
-            || (config.mode() != expectedConfig.mode())) {
+        if (config.mode() != expectedConfig.mode()) {
           throw new Exception(
-              "[FAILED] Expected: expected duration : "
-                  + expectedConfig.duration()
-                  + ", got: "
-                  + config.duration()
-                  + " expected mode :"
-                  + expectedConfig.mode()
-                  + ", got: "
-                  + config.mode());
+              "[FAILED] mode: expected: " + expectedConfig.mode() + ", got: " + config.mode());
         }
+
+        if (config.duration().unit() != expectedConfig.duration().unit()
+            || config.duration().duration() != expectedConfig.duration().duration()) {
+          throw new Exception(
+              "[FAILED] duration: " + expectedConfig.duration() + ", got: " + config.duration());
+        }
+
       } finally {
         client.removeBucket(RemoveBucketArgs.builder().bucket(bucketName).build());
       }
 
-      mintSuccessLog("getDefaultRetention (String bucketName)", null, startTime);
-
+      mintSuccessLog(methodName, null, startTime);
```

Comment:
The retention mode and duration can be logged as arguments

Reply:
It wouldn't be useful because the failure may happen any one of two checks. I am not favour of mint args because stack trace is the best to find the issue.

Reply:
Actually if the `try-catch` and exception handling is also moved to the new method (`testGetDefaultRetention`) then the args logging can be done.